File: app/src/main/java/com/returnnotfound/stain/data/repository/AnimeRepositoryImpl.java
package com.returnnotfound.stain.data.repository;

import com.returnnotfound.stain.data.dto.AnimeTopWraperDTO;
import com.returnnotfound.stain.data.mapper.AnimeTopWraperMapper;
import com.returnnotfound.stain.data.model.AnimeTop;
import com.returnnotfound.stain.data.remote.AnimeService;

import java.util.List;

import javax.inject.Inject;

import io.reactivex.Single;

public class AnimeRepositoryImpl implements AnimeRepository {
  private AnimeService mAnimeService;

  @Inject
  public AnimeRepositoryImpl(AnimeService animeService) {
    this.mAnimeService = animeService;
  }

  @Override
  public Single<List<AnimeTop>> getAnimeTop(int page, String subType) {
    return mAnimeService.getTop(page, subType)
        .map((AnimeTopWraperDTO animeTopWraperDTO) -> AnimeTopWraperMapper.INSTANCE.toModel(animeTopWraperDTO).getTopList());
  }
}

File: app/src/main/java/com/returnnotfound/stain/data/repository/AnimeRepositoryImpl.kt
package com.returnnotfound.stain.data.repository

import com.returnnotfound.stain.data.dto.AnimeTopWrapperDTO
import com.returnnotfound.stain.data.mapper.AnimeTopWrapperMapper
import com.returnnotfound.stain.data.model.AnimeTop
import com.returnnotfound.stain.data.remote.AnimeService
import io.reactivex.Single
import javax.inject.Inject

class AnimeRepositoryImpl @Inject constructor(private val mAnimeService: AnimeService) :
  AnimeRepository {

  override fun getAnimeTop(
    page: Int,
    subType: String?
  ): Single<List<AnimeTop?>?>? {
    return mAnimeService.getTop(page, subType)?.map { animeTopWrapperDTO: AnimeTopWrapperDTO? ->
      AnimeTopWrapperMapper.INSTANCE.toModel(animeTopWrapperDTO)?.topList
    }
  }

}